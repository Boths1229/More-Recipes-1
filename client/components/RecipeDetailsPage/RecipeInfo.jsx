import React, { PropTypes } from 'react';
import Tooltip from 'react-tooltip';

const propTypes = {
  recipeDetails: PropTypes.shape({
    recipeName: PropTypes.string,
    description: PropTypes.string,
    ingredients: PropTypes.string,
    method: PropTypes.string,
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
    image: PropTypes.string,
  }).isRequired,
  upvote: PropTypes.func.isRequired,
  downvote: PropTypes.func.isRequired,
  favorite: PropTypes.func.isRequired,
  favorited: PropTypes.bool.isRequired,
};

/* eslint-disable react/no-array-index-key  */

const RecipeInfo = props => (
  <div>
    <img className="card-img-top" src={props.recipeDetails.image} alt="Card pic cap" style={{ maxHeight: '350px' }} />
    <div className="card-body">
      <h4 className="card-title">{props.recipeDetails.recipeName}</h4>
      <p className="card-text">
        {props.recipeDetails.description}
      </p>
      <hr />
      <div>
        <h6>Ingredients</h6>
        <ul>
          <div className="row">
            {
              props.recipeDetails.ingredients.split(',').map((ingredients, i) => (
                <li
                  className="col-md-6"
                  style={{ paddingTop: '5px', fontSize: '18px' }}
                  key={i}
                >
                  {ingredients}
                </li>
              ))
            }
          </div>

        </ul>
      </div>
      <hr />
      <div>
        <h6>
          Direction
        </h6>
        <ol>
          {
            props.recipeDetails.method.split('\n').map((method, i) => (
              <li key={i}>
                {method}
              </li>
            ))
          }
        </ol>
      </div>
      <hr />
      <div style={{ display: 'inline-flex' }}>
        <div>
          <button
            className="btn vote-button"
            type="button"
            onClick={props.upvote}
            data-tip
            data-for="upvote"
          >
            <i className="fa fa-thumbs-up fa-lg" aria-hidden="true">
              <span style={{ marginLeft: '5px', marginRight: '5px' }}>
                {props.recipeDetails.upvotes}
              </span>
            </i>
          </button>
          <Tooltip
            id="upvote"
            type="dark"
            place="top"
          >
            <span>
              upvote recipe
            </span>
          </Tooltip>
        </div>
        <div style={{ marginLeft: '8px', marginRight: '5px' }}>
          <button
            data-tip
            data-for="downvote"
            className="btn vote-button"
            type="button"
            onClick={props.downvote}
          >
            <i className="fa fa-thumbs-down fa-lg" aria-hidden="true">
              <span style={{ marginLeft: '5px', marginRight: '5px' }}>
                {props.recipeDetails.downvotes}
              </span>
            </i>
          </button>
          <Tooltip
            id="downvote"
            type="dark"
            place="bottom"
          >
            <span>
              downvote recipe
            </span>
          </Tooltip>
        </div>
        <div style={{ marginLeft: '8px', marginRight: '5px' }}>
          <button
            className="btn vote-button"
            type="button"
            onClick={props.favorite}
            data-tip
            data-for="favorite"
          >
            <i
              className="fa fa-heart"
              style={
                props.favorited ? { color: 'white' } : { color: 'black' }
              }
            />
          </button>
          <Tooltip
            id="favorite"
            type="dark"
            place="right"
          >
            <span>
              favorite recipe
            </span>
          </Tooltip>
        </div>
      </div>
      <hr />
    </div>
  </div>
);

RecipeInfo.propTypes = propTypes;

export default RecipeInfo;
