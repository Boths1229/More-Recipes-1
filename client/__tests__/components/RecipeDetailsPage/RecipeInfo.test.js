import React from 'react';
import { shallow } from 'enzyme';
import RecipeInfo from '../../../components/RecipeDetailsPage/RecipeInfo';
import MockData from '../../__mocks__/actions/recipes';

let props;

const setup = () => {
  props = {
    recipeDetails: MockData.recipeData,
    upvote: () => {},
    downvote: () => {},
    favorite: () => {},
    favorited: true,
  };

  return shallow(<RecipeInfo {...props} />);
};

describe('RecipeInfo', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h6').length).toBe(3);
    expect(wrapper.find('button').length).toBe(3);
  });
});
