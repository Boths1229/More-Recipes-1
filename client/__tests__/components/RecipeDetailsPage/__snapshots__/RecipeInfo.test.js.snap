// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeInfo should render correctly 1`] = `
<div>
  <img
    alt="Card pic cap"
    className="card-img-top"
    src="https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212"
    style={
      Object {
        "maxHeight": "350px",
      }
    }
  />
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      new recipe
    </h4>
    <p
      className="card-text"
    >
      amazing
    </p>
    <hr />
    <div>
      <h6>
        Type of Meal
      </h6>
      <ul>
        <li>
           
          breakfast
           
        </li>
      </ul>
    </div>
    <hr />
    <div>
      <h6>
        Ingredients
      </h6>
      <ul>
        <div
          className="row"
        >
          <li
            className="col-md-6"
            key="0"
            style={
              Object {
                "fontSize": "18px",
                "paddingTop": "5px",
              }
            }
          >
            awesome
          </li>
        </div>
      </ul>
    </div>
    <hr />
    <div>
      <h6>
        Method of Cooking
      </h6>
      <ol>
        <li
          key="0"
        >
          great
        </li>
      </ol>
    </div>
    <hr />
    <div
      style={
        Object {
          "display": "inline-flex",
        }
      }
    >
      <div>
        <button
          className="btn vote-button"
          data-for="upvote"
          data-tip={true}
          id="upvote-button"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="fa fa-thumbs-up fa-lg"
          >
            <span
              style={
                Object {
                  "marginLeft": "5px",
                  "marginRight": "5px",
                }
              }
            >
              0
            </span>
          </i>
        </button>
        <ReactTooltip
          id="upvote"
          insecure={true}
          place="top"
          resizeHide={true}
          type="dark"
          wrapper="div"
        >
          <span>
            upvote recipe
          </span>
        </ReactTooltip>
      </div>
      <div
        style={
          Object {
            "marginLeft": "8px",
            "marginRight": "5px",
          }
        }
      >
        <button
          className="btn vote-button"
          data-for="downvote"
          data-tip={true}
          id="downvote-button"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-hidden="true"
            className="fa fa-thumbs-down fa-lg"
          >
            <span
              style={
                Object {
                  "marginLeft": "5px",
                  "marginRight": "5px",
                }
              }
            >
              1
            </span>
          </i>
        </button>
        <ReactTooltip
          id="downvote"
          insecure={true}
          place="bottom"
          resizeHide={true}
          type="dark"
          wrapper="div"
        >
          <span>
            downvote recipe
          </span>
        </ReactTooltip>
      </div>
      <div
        style={
          Object {
            "marginLeft": "8px",
            "marginRight": "5px",
          }
        }
      >
        <button
          className="btn vote-button"
          data-for="favorite"
          data-tip={true}
          id="favoriteButton"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-heart"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </button>
        <ReactTooltip
          id="favorite"
          insecure={true}
          place="right"
          resizeHide={true}
          type="dark"
          wrapper="div"
        >
          <span>
            favorite recipe
          </span>
        </ReactTooltip>
      </div>
    </div>
    <hr />
  </div>
</div>
`;
